gossip/gossip.go

```diff
@@ -502,12 +502,12 @@ func (g *Gossip) getNodeDescriptorLocked(nodeID roachpb.NodeID) (*roachpb.NodeDe
 // assumed held by the caller. This method is called externally via
 // GetNodeIDAddress or internally when looking up a "distant" node address to
 // connect directly to.
-func (g *Gossip) getNodeIDAddressLocked(nodeID roachpb.NodeID) (net.Addr, error) {
+func (g *Gossip) getNodeIDAddressLocked(nodeID roachpb.NodeID) (*util.UnresolvedAddr, error) {
 	nd, err := g.getNodeDescriptorLocked(nodeID)
 	if err != nil {
 		return nil, err
 	}
-	return nd.Address, nil
+	return &nd.Address, nil
```

Comment:
Because some callers want a pointer (e.g. `rpcTransport.processQueue`). Nobody is mutating these things and you'll notice that these pointers are newly allocated on every call.